invenio_rdm_records/resources/config.py

```diff
@@ -177,6 +178,17 @@ class RDMRecordResourceConfig(RecordResourceConfig, ConfiguratorMixin):
                 description=e.description,
             )
         ),
+        RecordDeletedException: create_error_handler(
+            lambda e: (
+                HTTPJSONException(code=404, description=_("Record not found"))
+                if not e.record.tombstone.is_visible
+                else HTTPJSONException(
+                    code=410,
+                    description=_("Record deleted"),
+                    tombstone=e.record.tombstone.dump(),
```

Comment:
if we want to keep other entries (like `links`, `identifier`, etc.) in the response, this is the place to put them for the REST API.